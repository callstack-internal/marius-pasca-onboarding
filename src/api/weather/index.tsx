import { useQuery } from "@tanstack/react-query";
import { API_KEY } from "@env"
import { WeatherData } from "./types";

export const WEATHER_API_URL = 'https://api.openweathermap.org/data/2.5/group';

const fetchGroupCities = async (cityIds: number[]): Promise<WeatherData[]> => {
    const cityIdsString = cityIds.join(',');
    const response = await fetch(`${WEATHER_API_URL}?id=${cityIdsString}&appid=${API_KEY}&units=imperial`);

    if (!response.ok) {
        throw new Error('Failed to fetch weather data');
    }

    const data = await response.json();
    return data.list;
};

export function useGroupWeather(cityIds: number[]) {
    return useQuery({ queryKey: ['groupWeather'], queryFn: () => fetchGroupCities(cityIds) });
}