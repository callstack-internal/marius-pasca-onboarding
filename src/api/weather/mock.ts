export const WEATHER_DATA = [
    {
        "coord": {
            "lon": 30.5167,
            "lat": 50.4333
        },
        "weather": [
            {
                "id": 804,
                "main": "Clouds",
                "description": "overcast clouds",
                "icon": "04d"
            }
        ],
        "base": "stations",
        "main": {
            "temp": 292.44,
            "feels_like": 291.47,
            "temp_min": 289.63,
            "temp_max": 292.44,
            "pressure": 1029,
            "humidity": 40,
            "sea_level": 1029,
            "grnd_level": 1013
        },
        "visibility": 10000,
        "wind": {
            "speed": 0.45,
            "deg": 229,
            "gust": 1.79
        },
        "clouds": {
            "all": 85
        },
        "dt": 1729599609,
        "sys": {
            "type": 2,
            "id": 2003742,
            "country": "UA",
            "sunrise": 1729571524,
            "sunset": 1729608742
        },
        "timezone": 10800,
        "id": 703448,
        "name": "Kyiv",
        "cod": 200
    },
    {
        "coord": {
            "lon": 34.8003,
            "lat": 50.9216
        },
        "weather": [
            {
                "id": 803,
                "main": "Clouds",
                "description": "broken clouds",
                "icon": "04d"
            }
        ],
        "base": "stations",
        "main": {
            "temp": 288.63,
            "feels_like": 287.1,
            "temp_min": 288.63,
            "temp_max": 288.63,
            "pressure": 1029,
            "humidity": 33,
            "sea_level": 1029,
            "grnd_level": 1009
        },
        "visibility": 10000,
        "wind": {
            "speed": 3.5,
            "deg": 269,
            "gust": 4.93
        },
        "clouds": {
            "all": 59
        },
        "dt": 1729599442,
        "sys": {
            "country": "UA",
            "sunrise": 1729570550,
            "sunset": 1729607659
        },
        "timezone": 10800,
        "id": 692194,
        "name": "Sumy",
        "cod": 200
    },
    {
        "coord": {
            "lon": 21.0118,
            "lat": 52.2298
        },
        "weather": [
            {
                "id": 800,
                "main": "Clear",
                "description": "clear sky",
                "icon": "01d"
            }
        ],
        "base": "stations",
        "main": {
            "temp": 290.54,
            "feels_like": 289.74,
            "temp_min": 289.76,
            "temp_max": 291.85,
            "pressure": 1024,
            "humidity": 54,
            "sea_level": 1024,
            "grnd_level": 1012
        },
        "visibility": 10000,
        "wind": {
            "speed": 4.92,
            "deg": 220,
            "gust": 0
        },
        "clouds": {
            "all": 0
        },
        "dt": 1729599713,
        "sys": {
            "type": 2,
            "id": 2032856,
            "country": "PL",
            "sunrise": 1729574019,
            "sunset": 1729610809
        },
        "timezone": 7200,
        "id": 756135,
        "name": "Warsaw",
        "cod": 200
    },
    {
        "coord": {
            "lon": 17.0333,
            "lat": 51.1
        },
        "weather": [
            {
                "id": 800,
                "main": "Clear",
                "description": "clear sky",
                "icon": "01d"
            }
        ],
        "base": "stations",
        "main": {
            "temp": 293.37,
            "feels_like": 292.91,
            "temp_min": 293.15,
            "temp_max": 293.94,
            "pressure": 1024,
            "humidity": 56,
            "sea_level": 1024,
            "grnd_level": 1008
        },
        "visibility": 10000,
        "wind": {
            "speed": 5.66,
            "deg": 240
        },
        "clouds": {
            "all": 0
        },
        "dt": 1729599727,
        "sys": {
            "type": 2,
            "id": 2006601,
            "country": "PL",
            "sunrise": 1729574840,
            "sunset": 1729611897
        },
        "timezone": 7200,
        "id": 3081368,
        "name": "Wrocław",
        "cod": 200
    },
    {
        "coord": {
            "lon": 14.4208,
            "lat": 50.088
        },
        "weather": [
            {
                "id": 801,
                "main": "Clouds",
                "description": "few clouds",
                "icon": "02d"
            }
        ],
        "base": "stations",
        "main": {
            "temp": 289.59,
            "feels_like": 289.35,
            "temp_min": 288.18,
            "temp_max": 292.6,
            "pressure": 1026,
            "humidity": 79,
            "sea_level": 1026,
            "grnd_level": 992
        },
        "visibility": 10000,
        "wind": {
            "speed": 3.09,
            "deg": 240
        },
        "clouds": {
            "all": 20
        },
        "dt": 1729599648,
        "sys": {
            "type": 1,
            "id": 6848,
            "country": "CZ",
            "sunrise": 1729575353,
            "sunset": 1729612638
        },
        "timezone": 7200,
        "id": 3067696,
        "name": "Prague",
        "cod": 200
    },
    {
        "coord": {
            "lon": 14.4743,
            "lat": 48.9745
        },
        "weather": [
            {
                "id": 800,
                "main": "Clear",
                "description": "clear sky",
                "icon": "01d"
            }
        ],
        "base": "stations",
        "main": {
            "temp": 290.1,
            "feels_like": 289.73,
            "temp_min": 288.75,
            "temp_max": 292.09,
            "pressure": 1027,
            "humidity": 72,
            "sea_level": 1027,
            "grnd_level": 970
        },
        "visibility": 10000,
        "wind": {
            "speed": 1.79,
            "deg": 211,
            "gust": 1.79
        },
        "clouds": {
            "all": 2
        },
        "dt": 1729599483,
        "sys": {
            "type": 2,
            "id": 2045921,
            "country": "CZ",
            "sunrise": 1729575219,
            "sunset": 1729612746
        },
        "timezone": 7200,
        "id": 3077916,
        "name": "České Budějovice",
        "cod": 200
    },
    {
        "coord": {
            "lon": 13.4105,
            "lat": 52.5244
        },
        "weather": [
            {
                "id": 803,
                "main": "Clouds",
                "description": "broken clouds",
                "icon": "04d"
            }
        ],
        "base": "stations",
        "main": {
            "temp": 287.58,
            "feels_like": 287.17,
            "temp_min": 286.53,
            "temp_max": 288.9,
            "pressure": 1025,
            "humidity": 80,
            "sea_level": 1025,
            "grnd_level": 1020
        },
        "visibility": 10000,
        "wind": {
            "speed": 8.23,
            "deg": 290
        },
        "clouds": {
            "all": 75
        },
        "dt": 1729599764,
        "sys": {
            "type": 2,
            "id": 2011538,
            "country": "DE",
            "sunrise": 1729575881,
            "sunset": 1729612594
        },
        "timezone": 7200,
        "id": 2950159,
        "name": "Berlin",
        "cod": 200
    },
    {
        "coord": {
            "lon": 11.5755,
            "lat": 48.1374
        },
        "weather": [
            {
                "id": 804,
                "main": "Clouds",
                "description": "overcast clouds",
                "icon": "04d"
            }
        ],
        "base": "stations",
        "main": {
            "temp": 290.56,
            "feels_like": 290.47,
            "temp_min": 289.15,
            "temp_max": 291.66,
            "pressure": 1029,
            "humidity": 81,
            "sea_level": 1029,
            "grnd_level": 965
        },
        "visibility": 10000,
        "wind": {
            "speed": 1.16,
            "deg": 232,
            "gust": 2.66
        },
        "clouds": {
            "all": 100
        },
        "dt": 1729599485,
        "sys": {
            "type": 2,
            "id": 2013241,
            "country": "DE",
            "sunrise": 1729575828,
            "sunset": 1729613529
        },
        "timezone": 7200,
        "id": 2867714,
        "name": "Munich",
        "cod": 200
    },
    {
        "coord": {
            "lon": 6.0834,
            "lat": 50.7766
        },
        "weather": [
            {
                "id": 803,
                "main": "Clouds",
                "description": "broken clouds",
                "icon": "04d"
            }
        ],
        "base": "stations",
        "main": {
            "temp": 287.38,
            "feels_like": 286.82,
            "temp_min": 286.55,
            "temp_max": 288.13,
            "pressure": 1030,
            "humidity": 75,
            "sea_level": 1030,
            "grnd_level": 1002
        },
        "visibility": 10000,
        "wind": {
            "speed": 4.76,
            "deg": 261,
            "gust": 6.08
        },
        "clouds": {
            "all": 54
        },
        "dt": 1729599677,
        "sys": {
            "type": 2,
            "id": 2013497,
            "country": "DE",
            "sunrise": 1729577434,
            "sunset": 1729614559
        },
        "timezone": 7200,
        "id": 3247449,
        "name": "Aachen",
        "cod": 200
    },
    {
        "coord": {
            "lon": -120.5015,
            "lat": 47.5001
        },
        "weather": [
            {
                "id": 800,
                "main": "Clear",
                "description": "clear sky",
                "icon": "01n"
            }
        ],
        "base": "stations",
        "main": {
            "temp": 276.27,
            "feels_like": 274.35,
            "temp_min": 274.62,
            "temp_max": 278.37,
            "pressure": 1025,
            "humidity": 77,
            "sea_level": 1025,
            "grnd_level": 938
        },
        "visibility": 10000,
        "wind": {
            "speed": 1.98,
            "deg": 271,
            "gust": 1.66
        },
        "clouds": {
            "all": 1
        },
        "dt": 1729599727,
        "sys": {
            "type": 2,
            "id": 2003593,
            "country": "US",
            "sunrise": 1729607494,
            "sunset": 1729645253
        },
        "timezone": -25200,
        "id": 5815135,
        "name": "Washington",
        "cod": 200
    },
    {
        "coord": {
            "lon": -74.006,
            "lat": 40.7143
        },
        "weather": [
            {
                "id": 800,
                "main": "Clear",
                "description": "clear sky",
                "icon": "01d"
            }
        ],
        "base": "stations",
        "main": {
            "temp": 290.01,
            "feels_like": 289.58,
            "temp_min": 287.27,
            "temp_max": 291.6,
            "pressure": 1024,
            "humidity": 70,
            "sea_level": 1024,
            "grnd_level": 1022
        },
        "visibility": 10000,
        "wind": {
            "speed": 5.81,
            "deg": 24,
            "gust": 6.71
        },
        "clouds": {
            "all": 0
        },
        "dt": 1729599406,
        "sys": {
            "type": 2,
            "id": 2037026,
            "country": "US",
            "sunrise": 1729595715,
            "sunset": 1729634717
        },
        "timezone": -14400,
        "id": 5128581,
        "name": "New York",
        "cod": 200
    }
]

export const DEFAULT_CITY_IDS = [
    703448, // Kyiv, UA
    692194, // Sumy, UA
    756135, // Warsaw, PL
    3081368, // Wrocław, PL
    3067696, // Prague, CZ
    3077916, // České Budějovice, CZ
    2950159, // Berlin, DE
    2867714, // Munich, DE
    3247449, // Aachen, DE
    5815135, // Washington, US
    5128581, // New York City, US
];