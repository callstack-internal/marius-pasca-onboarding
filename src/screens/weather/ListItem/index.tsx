import React from 'react';
import { View, StyleSheet, TouchableOpacity, Image } from 'react-native';
import chevron_right from '@assets/images/chevron_right.png';
import { WeatherData } from '@api/weather/types';
import CityWeather from '@components/CityWeather';
import { theme } from '@styles/index';

type Props = {
    navigateToDetails: () => void;
    data: WeatherData;
}

const WeatherListItem: React.FC<Props> = ({ navigateToDetails, data }) => {
    return (
        <TouchableOpacity onPress={navigateToDetails} style={styles.container} testID='navigat_to_details'>
            <View style={styles.cityContainer}>
                <CityWeather data={data} />
            </View>
            <View style={styles.iconContainer}>
                <Image source={chevron_right} style={styles.rightIcon} testID='chevron_right' />
            </View>
        </TouchableOpacity>
    )
}

const styles = StyleSheet.create({
    container: {
        backgroundColor: theme.colors.WHITE,
        flexDirection: 'row',
        justifyContent: 'space-between',
        borderBottomWidth: 1,
        borderBottomColor: theme.colors.GREY_LIGHT,
    },
    cityContainer: {
        flex: 1,
    },
    iconContainer: {
        justifyContent: 'center',
    },
    rightIcon: {
        width: 50,
        height: 50,
    },
});

export default WeatherListItem;
