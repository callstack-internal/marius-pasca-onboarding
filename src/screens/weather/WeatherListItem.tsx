import React from 'react';
import { View, StyleSheet, TouchableOpacity, Image } from 'react-native';
import chevron_right from '@assets/images/chevron_right.png';
import { WeatherData } from '@api/weather/types';
import CityWeather from '@components/CityWeather';

type Props = {
    navigateToDetails: () => void;
    data: WeatherData;
}

const WeatherListItem: React.FC<Props> = ({ navigateToDetails, data }) => {
    return (
        <TouchableOpacity onPress={navigateToDetails} style={styles.container} testID='navigat_to_details'>
            <View style={styles.cityContainer}>
                <CityWeather data={data} />
            </View>
            <View style={styles.iconContainer}>
                <Image source={chevron_right} style={styles.rightIcon} testID='chevron_right' />
            </View>
        </TouchableOpacity>
    )
}

const styles = StyleSheet.create({
    container: {
        backgroundColor: 'white',
        flexDirection: 'row',
        justifyContent: 'space-between',
        borderBottomWidth: 1,
        borderBottomColor: '#ccc',
    },
    cityContainer: {
        flex: 1,
    },
    iconContainer: {
        justifyContent: 'center',
    },
    rightIcon: {
        width: 50,
        height: 50,
    },
});

export default WeatherListItem;
