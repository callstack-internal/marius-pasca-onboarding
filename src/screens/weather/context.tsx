import { useGroupWeather } from "@api/weather";
import { DEFAULT_CITY_IDS } from "@api/weather/mock";
import { WeatherData } from "@api/weather/types";
import { UseQueryResult } from "@tanstack/react-query";
import { createContext, useMemo, useState, PropsWithChildren } from "react";

export const WeatherContext = createContext<{
    search?: string,
    setSearch?: (search: string) => void,
    weatherQuery?: UseQueryResult<WeatherData[], Error>;
}>({});

const filterCities = (cities: { [city: string]: number }, searchText: string) => {
    return Object.fromEntries(
        Object.entries(cities).filter(([city]) =>
            city.toLowerCase().includes(searchText.toLowerCase())
        )
    );
};

export const WeatherProvider: React.FC<PropsWithChildren> = ({ children }) => {
    const [search, setSearch] = useState('');

    const cities = useMemo(() => {
        const citiesObj = search ? filterCities(DEFAULT_CITY_IDS, search) : DEFAULT_CITY_IDS;
        return Object.values(citiesObj)
    }, [search]);

    const weatherQuery = useGroupWeather(cities);
    const data = weatherQuery.data?.filter((city) => cities.includes(city.id));

    return (
        <WeatherContext.Provider value={{ search, setSearch, weatherQuery: { ...weatherQuery, data } as UseQueryResult<WeatherData[], Error> }}>
            {children}
        </WeatherContext.Provider>
    );
};