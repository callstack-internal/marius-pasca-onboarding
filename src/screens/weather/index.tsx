import React from 'react';
import { View, Text, FlatList, ActivityIndicator, StyleSheet, ListRenderItem } from 'react-native';
import { NativeStackScreenProps } from '@react-navigation/native-stack';
import { RootStackParamList } from '../../navigation/types';
import { useGroupWeather } from '../../api/weather';
import { WeatherData } from '../../api/weather/types';
import WeatherListItem from './WeatherListItem';

type Props = NativeStackScreenProps<RootStackParamList, 'Weather'>;

const WeatherList: React.FC<Props> = ({ navigation }) => {
  const { data, isLoading, error, refetch } = useGroupWeather();

  if (isLoading) return <View style={styles.placeholder}>
    <ActivityIndicator size="large" color="#f9794e" />
  </View>;

  if (error) return <View style={styles.placeholder}>
    <Text style={styles.text}>Error fetching data.</Text>
  </View>;

  const renderItem: ListRenderItem<WeatherData> = ({ item }) => {
    const navigateToDetails = () => {
      navigation.navigate('Details', { data: item });
    }

    return <WeatherListItem data={item} navigateToDetails={navigateToDetails} />
  };

  const emptyComponent = () => <View style={styles.placeholder}>
    <Text style={styles.text}>There is no data available.</Text>
  </View>;

  return (
    <View style={styles.container}>
      <FlatList
        data={data}
        keyExtractor={(item) => item.id.toString()}
        renderItem={renderItem}
        onRefresh={refetch}
        ListEmptyComponent={emptyComponent}
        refreshing={isLoading}
      />
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
  },
  placeholder: {
    padding: 20,
  },
  text: {
    fontSize: 18,
    fontWeight: '700',
    color: '#909090',
  }
});

export default WeatherList;
