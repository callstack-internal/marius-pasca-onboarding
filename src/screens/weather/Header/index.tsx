import { theme } from '@styles/index';
import React, { useContext, useEffect, useState } from 'react';
import searchIcon from '@assets/images/search.png';
import { View, TextInput, StyleSheet, TouchableOpacity, Animated, Easing, Text, Image } from 'react-native';
import { useSafeAreaInsets } from 'react-native-safe-area-context';
import { WeatherContext } from '../context';
import useDebounced from '@components/utils/useDebounced';

const Header = () => {
    const { top } = useSafeAreaInsets();
    const [isExpanded, setIsExpanded] = useState(false);
    const { setSearch } = useContext(WeatherContext);
    const [searchText, setSearchText] = useState('');
    const heightAnim = useState(new Animated.Value(0))[0];

    const debouncedSearch = useDebounced(searchText, 500);

    const toggleExpand = () => {
        setIsExpanded((prev) => !prev);
        Animated.timing(heightAnim, {
            toValue: isExpanded ? 0 : 40,
            duration: 200,
            easing: Easing.ease,
            useNativeDriver: false,
        }).start();
    };

    useEffect(() => {
        setSearch?.(debouncedSearch);
    }, [setSearch, debouncedSearch]);

    const onChangeText = (text: string) => {
        setSearchText(text);
    };

    return (
        <View style={[styles.container, { paddingTop: top }]}>
            <View style={styles.titleContainer}>
                <Text style={styles.title}>Weather</Text>
                <TouchableOpacity onPress={toggleExpand} style={styles.searchContainer}>
                    {Boolean(searchText) && <View style={styles.bullet} />}
                    <Image source={searchIcon} style={styles.searchIcon} />
                </TouchableOpacity>
            </View>
            <Animated.View style={[styles.searchBar, { height: heightAnim }]}>
                {isExpanded && <TextInput
                    style={styles.input}
                    placeholder="Search..."
                    value={searchText}
                    onChangeText={onChangeText}
                    autoFocus
                    onBlur={toggleExpand}
                />}
            </Animated.View>
        </View>
    );
};

const styles = StyleSheet.create({
    container: {
        backgroundColor: theme.colors.PRIMARY,
        padding: 10,
        paddingBottom: 0,
    },
    titleContainer: {
        height: 28,
        flexDirection: 'row',
        alignItems: 'center',
        justifyContent: 'center',
    },
    title: {
        fontSize: 18,
        fontWeight: '700',
        color: theme.colors.WHITE,
    },
    searchContainer: {
        position: 'absolute',
        backgroundColor: theme.colors.WHITE,
        borderRadius: 10,
        right: 10,
    },
    searchIcon: {
        width: 32,
        height: 32,
    },
    searchBar: {
        backgroundColor: theme.colors.WHITE,
        borderRadius: 10,
        marginHorizontal: 10,
        marginVertical: 5,
    },
    bullet: {
        height: 10,
        width: 10,
        borderRadius: 10,
        position: 'absolute',
        right: -2,
        top: -2,
        backgroundColor: theme.colors.SECONDARY,
    },
    input: {
        flex: 1,
        marginLeft: 16,
        fontSize: 16,
    },
    icon: {
        marginLeft: 5,
    },
});

export default Header;

