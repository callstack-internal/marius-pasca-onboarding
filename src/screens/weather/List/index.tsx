import React, { useContext } from 'react';
import { View, Text, FlatList, ActivityIndicator, StyleSheet, ListRenderItem } from 'react-native';
import { NavigationTypes } from '@navigation/types';
import { WeatherData } from '@api/weather/types';
import WeatherListItem from '../ListItem';
import { useSafeAreaInsets } from 'react-native-safe-area-context';
import { theme } from '@styles/index';
import Header from '../Header';
import { WeatherContext } from '../context';
import { useNavigation } from '@react-navigation/native';

type Props = {};

const WeatherList: React.FC<Props> = () => {
    const navigation = useNavigation<NavigationTypes>();
    const { bottom } = useSafeAreaInsets();
    const { weatherQuery } = useContext(WeatherContext);
    const { isLoading, error, data, refetch } = weatherQuery || {};

    if (isLoading) return <View style={styles.placeholder}>
        <ActivityIndicator size="large" color={theme.colors.PRIMARY} testID='activity-indicator' />
    </View>;

    if (error) return <View style={styles.placeholder}>
        <Text style={styles.text}>Error fetching data.</Text>
    </View>;

    const renderItem: ListRenderItem<WeatherData> = ({ item }) => {
        const navigateToDetails = () => {
            navigation.navigate('Details', { data: item });
        }

        return <WeatherListItem data={item} navigateToDetails={navigateToDetails} />
    };

    const emptyComponent = () => <View style={styles.placeholder}>
        <Text style={styles.text}>There is no data available.</Text>
    </View>;

    return (
        <View style={styles.container}>
            <Header />
            <FlatList
                data={data}
                keyExtractor={(item) => item.id.toString()}
                renderItem={renderItem}
                onRefresh={refetch}
                ListEmptyComponent={emptyComponent}
                refreshing={isLoading}
                ListFooterComponent={<View style={{ paddingBottom: bottom }} />}
            />
        </View>
    );
};

const styles = StyleSheet.create({
    container: {
        flex: 1,
    },
    placeholder: {
        padding: 20,
    },
    text: {
        fontSize: 18,
        fontWeight: '700',
        color: theme.colors.GREY,
    }
});

export default WeatherList;
