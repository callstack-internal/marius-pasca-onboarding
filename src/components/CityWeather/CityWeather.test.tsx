import React from 'react';
import { render } from '@testing-library/react-native';
import CityWeather from '.';
import { WEATHER_DATA } from '@api/weather/mock';

jest.mock('@assets/images/clear_sky.png', () => 'clear_sky.png');
jest.mock('@assets/images/few_clouds.png', () => 'few_clouds.png');
jest.mock('@assets/images/scattered_clouds.png', () => 'scattered_clouds.png');
jest.mock('@assets/images/broken_clouds.png', () => 'broken_clouds.png');
jest.mock('@assets/images/shower_rain.png', () => 'shower_rain.png');
jest.mock('@assets/images/rain.png', () => 'rain.png');
jest.mock('@assets/images/thunderstorm.png', () => 'thunderstorm.png');
jest.mock('@assets/images/snow.png', () => 'snow.png');
jest.mock('@assets/images/mist.png', () => 'mist.png');

describe("CityWeather Component", () => {
    it("renders correctly and matches snapshot", () => {
        const { toJSON } = render(<CityWeather data={WEATHER_DATA[0]} />);
        expect(toJSON()).toMatchSnapshot();
    });
});
