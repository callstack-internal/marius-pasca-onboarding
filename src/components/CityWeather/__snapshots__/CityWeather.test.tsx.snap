// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CityWeather Component renders correctly and matches snapshot 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flexDirection": "row",
      "justifyContent": "space-between",
      "padding": 10,
    }
  }
  testID="city_weather_component"
>
  <View
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "gap": 10,
      }
    }
  >
    <Image
      source="broken_clouds.png"
      style={
        {
          "height": 70,
          "width": 70,
        }
      }
    />
    <View>
      <Text
        style={
          {
            "fontSize": 18,
            "fontWeight": "500",
          }
        }
      >
        Kyiv
      </Text>
      <Text
        style={
          {
            "color": "#909090",
            "fontSize": 14,
            "fontWeight": "700",
          }
        }
      >
        Overcast Clouds
      </Text>
    </View>
  </View>
  <View
    style={
      {
        "backgroundColor": "#6aa9b8",
        "borderRadius": 50,
        "paddingHorizontal": 15,
        "paddingVertical": 5,
      }
    }
  >
    <Text
      style={
        {
          "color": "white",
          "fontSize": 18,
          "fontWeight": "bold",
        }
      }
    >
      292.4
      Â°F
    </Text>
  </View>
</View>
`;
